caluma/caluma_form/tests/test_document.py

```diff
@@ -740,11 +743,22 @@ def test_save_document_answer(
         # delete answer to force create test instead of update
         Answer.objects.filter(pk=answer.pk).delete()
 
-    result = schema_executor(query, variable_values=inp)
+    if not use_api:
+        result = schema_executor(query, variable_values=inp)
+
+        assert not bool(result.errors) == success
 
-    assert not bool(result.errors) == success
-    if success:
-        snapshot.assert_match(result.data)
+        if success:
+            snapshot.assert_match(result.data)
+    else:
+        try:
+            answer = api.save_answer(
+                question, answer.document, user=admin_user, value=inp["input"]["value"]
+            )
+            snapshot.assert_match(answer)
+        except Exception:
+            if success:
```

Comment:
I'd prefer using `pytest.raises(TheExpectedException)`

Reply:
Hmm it's not really that we want to check *which* exception occurs, but rather *if* it occurs when we expect success. But I agree it's not so readable.